src/SPC/doctor/item/LinuxMuslCheck.php

```diff
@@ -8,57 +8,122 @@
-            default => CheckResult::fail('musl-libc is not installed on your system'),
-        };
+            $prefix = '';
+            if (get_current_user() !== 'root') {
+                $prefix = 'sudo ';
+                logger()->warning('Current user is not root, using sudo for running command');
+            }
+            $profile = '/etc/profile';
+            if (file_exists('~/.bash_profile')) {
+                $profile = '~/.bash_profile';
+            } elseif (file_exists('~/.profile')) {
+                $profile = '~/.profile';
+            }
+            if (!file_exists($profile)) {
+                shell(true)->exec($prefix . 'touch ' . $profile);
+            }
+            if (!getenv('PATH') || !str_contains(getenv('PATH'), '/usr/local/musl/bin')) {
+                $fix_path = 'echo "export PATH=/usr/local/musl/bin:$PATH" >> ' . $profile . ' && export PATH=/usr/local/musl/bin:$PATH';
+                shell(true)->exec($prefix . $fix_path);
+            }
+            return true;
+        } catch (RuntimeException) {
+            return false;
+        }
     }
 
     /**
+     * @throws DownloaderException
      * @throws RuntimeException
-     * @noinspection PhpUnused
+     * @throws FileSystemException
      */
-    #[AsFixItem('fix-musl')]
-    public function fixMusl(array $distro): bool
+    public function installMuslWrapper(): void
     {
-        $rhel_install = 'wget https://musl.libc.org/releases/musl-1.2.4.tar.gz && tar -zxvf musl-1.2.4.tar.gz && \
-                         rm -f musl-1.2.4.tar.gz && cd musl-1.2.4 && 
-                         if [[ ! "$PATH" =~ (^|:)"/usr/local/musl/bin"(:|$) ]]; then echo "export PATH=/usr/local/musl/bin:$PATH" >> ~/.bash_profile
-                         fi && \
-                         ./configure --enable-wrapper=gcc && \
-                         make -j && make install && cd .. && rm -rf musl-1.2.4';
-        $install_cmd = match ($distro['dist']) {
-            'ubuntu', 'debian' => 'apt-get install musl musl-tools -y',
-            'alpine' => 'apk add musl musl-utils musl-dev',
-            'rhel' => $rhel_install,
-            'almalinux' => $rhel_install,
-            default => throw new RuntimeException('Current linux distro does not have an auto-install script for musl packages yet.'),
-        };
         $prefix = '';
         if (get_current_user() !== 'root') {
             $prefix = 'sudo ';
             logger()->warning('Current user is not root, using sudo for running command');
         }
-        try {
-            shell(true)->exec($prefix . $install_cmd);
-            return true;
-        } catch (RuntimeException) {
-            return false;
-        }
+        $musl_source = [
+            'type' => 'url',
+            'url' => 'https://musl.libc.org/releases/musl-1.2.4.tar.gz',
+        ];
+        Downloader::downloadSource('musl-1.2.4', $musl_source);
+        FileSystem::extractSource('musl-1.2.4', DOWNLOAD_PATH . '/musl-1.2.4.tar.gz');
```

Comment:
Sounds counter-intuitive, but we should keep this version hard-coded as it should match the version that the musl-cross-make was built with!

Maybe add a comment there?